src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -34,6 +33,7 @@ public class MongoClientSettings : IEquatable<MongoClientSettings>, IInheritable
         // private fields
         private string _applicationName;
         private Action<ClusterBuilder> _clusterConfigurator;
+        private IEnumerable<ConnectionStringCompressorDescription> _compressors;
```

Comment:
I left it as it implemented, taking into account how we implemented the `Servers` property.
Also I'm not sure that we need to have `IReadOnlyList` here since this collection is not `readonly` from business point of view. We may want to add new compressors during `MongoClient` creating.
I guess here it's better to have `IList<>`? or `List<>`?

Reply:
See earlier comment about moving towards immutability.

It may be true that Servers is IEnumerable, but that seems like a mistake in hindsight. Do we want to repeat the mistake?